src/widgets/survey/model/hooks/useFlowStateActions.ts

```diff
@@ -128,11 +137,22 @@ export function useFlowStateActions({
     }
   }
 
-  function completeByTimer() {
+  function completeByTimer(): TimerCallbackFeedback {
     const record: FlowState = getCurrentFlowStorageRecord()!;
 
+    Logger.log(
+      `[useFlowStateActions.completeByTimer] Executing, current step is: ${record.step}`,
+    );
+
     if (isLastStep(record) || isSummaryStep(record)) {
-      return;
+      return 'no-feedback';
+    }
+
+    if (isInterimStep(record) && InterimSubmitObservable.processing) {
+      Logger.log(
+        `[useFlowStateActions.completeByTimer] Requested for postpone`,
+      );
+      return 'request-for-postpone';
```

Comment:
Please move the logic out of the function. It should be able to do both: completeByTimer and check `isInterimStep(record) && InterimSubmitObservable.processing`

Reply:
We already have similar logic in this handler. We return in case if we're on Summary on on Finish screen. Please specify.

Reply:
The function `completeByTimer` has a single responsibility (to upsert the flow storage record with values) that should not be violated.

Reply:
Resolved by Postponer